builder/main.py

```
@@ -1908,6 +1908,170 @@ def download_fs_action(target, source, env):
+
+    A .clang-format file in the project root is automatically respected.
+
+    Args:
+        force_mode: If set, overrides the mode from platformio.ini ("check" or "write")
+
+    Returns:
+        int: 0 on success, non-zero on error
+    """
+    project_dir = Path(get_project_dir())
+
+    # Resolve clang-format executable from platform package – install on demand
```

Ruff (0.15.2)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)